combustion-android-ble/src/main/java/inc/combustion/framework/ble/device/NodeBleDevice.kt

```diff
@@ -77,31 +78,37 @@ internal class NodeBleDevice(
         }
     }
 
-    private fun observeUartResponses(callback: (suspend (responses: List<NodeUARTMessage>) -> Unit)? = null) {
+    private fun observeUartMessages(callback: (suspend (messages: List<NodeUARTMessage>) -> Unit)? = null) {
         uart.jobManager.addJob(
             key = uart.serialNumber,
             job = uart.owner.lifecycleScope.launch(
                 CoroutineName("UartResponseObserver"),
             ) {
                 uart.observeUartCharacteristic {data ->
-                    val responses = NodeUARTMessage.fromData(data.toUByteArray())
-                    callback?.invoke(responses)
+                    val messages = NodeUARTMessage.fromData(data.toUByteArray())
+                    callback?.invoke(messages)
                 }
             }
         )
     }
-    private fun processUartResponses() {
-        observeUartResponses { responses ->
-            responses.forEach { response ->
-                when (response) {
+    private fun processUartMessages() {
+        observeUartMessages { messages ->
+            messages.forEach { message ->
+                when (message) {
                     is NodeReadFeatureFlagsResponse -> {
-                        readFeatureFlagsRequest.handled(response.success, response, response.requestId)
+                        readFeatureFlagsRequest.handled(message.success, message, message.requestId)
                     }
                     is GenericNodeResponse -> {
-                        genericRequestHandler.handled(response.success, response, response.requestId)
-                    } else -> {
+                        genericRequestHandler.handled(message.success, message, message.requestId)
+                    }
+                    is GenericNodeRequest -> {
+                        // Publish the request to the flow so it can be handled by the user.
+                        message.nodeSerialNumber = deviceInfoSerialNumber?: ""
```

Comment:
➕

Reply:
It will still emit the message that was received, just without that bit of information. If it gets to the app without this it can be dropped there. Are you suggesting it should be dropped here instead? I don't have a strong opinion either way and its not clear to me if one approach is better than the other.

Reply:
This will be a moot point if the serial number is added to the message.

Reply:
just checking if it was thought through - do not feel strongly about null vs "".